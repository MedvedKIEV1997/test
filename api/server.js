const express = require("express");

const app = express();

const PORT = 8000;

const employees = [
  {
    name: "Doug Lime",
    controls: ["Lin Cheng", "Bob Welp"],
  },
  {
    name: "Lin Cheng",
    controls: ["Dakota Stein", "Mongo K", "Dew Dong"],
  },
  {
    name: "Bob Welp",
    controls: ["Mort Dog", "Forman Go", "Dew Kelg"],
  },
  {
    name: "Dakota Stein",
    controls: [],
  },
  {
    name: "Mongo K",
    controls: [],
  },
  {
    name: "Dew Dong",
    controls: [],
  },
  {
    name: "Mort Dog",
    controls: [],
  },
  {
    name: "Forman Go",
    controls: [],
  },
  {
    name: "Dew Kelg",
    controls: [],
  },
];

app.get("/employees", (req, res) => {
  res.json(employees);
});

app.get("/employees/:name", (req, res) => {
  const name = req.params.name;
  const employee = employees.find((el) =>
    el.name.toLocaleLowerCase().includes(name.toLocaleLowerCase())
  );
  res.json(employee);
});

app.listen(PORT, () => {
  console.log("running");
});
