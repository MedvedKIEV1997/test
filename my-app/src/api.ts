import axios from 'axios';

export const url = 'http://localhost:8000/employees';

export const fetchEmployee = async (id: number) => {
    try {
        return await axios.get(`${url}/${id}`);
    } catch (error) {
        throw error;
    }
};

export const fetchEmployees = async () => {
    try {
        return await axios.get(url);
    } catch (error) {
        throw error;
    }
};
