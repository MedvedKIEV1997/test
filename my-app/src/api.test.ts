import { fetchEmployee, fetchEmployees } from './api';

describe('api testing', () => {
    test('fetchEmployees', async () => {
        const { data } = await fetchEmployees();
        expect(data).toMatchSnapshot();
    });
    test('fetchEmployee returns id 1', async () => {
        const { data } = await fetchEmployee(1);
        expect(data).toMatchSnapshot();
    });
});
