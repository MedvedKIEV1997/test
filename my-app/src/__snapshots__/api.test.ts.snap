// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api testing fetchEmployee returns id 1 1`] = `
Object {
  "controls": Array [
    2,
    3,
  ],
  "id": 1,
  "name": "Doug Lime",
}
`;

exports[`api testing fetchEmployees 1`] = `
Array [
  Object {
    "controls": Array [
      2,
      3,
    ],
    "id": 1,
    "name": "Doug Lime",
  },
  Object {
    "controls": Array [
      4,
      5,
      6,
    ],
    "id": 2,
    "name": "Lin Cheng",
  },
  Object {
    "controls": Array [
      7,
      8,
      9,
    ],
    "id": 3,
    "name": "Bob Welp",
  },
  Object {
    "controls": Array [],
    "id": 4,
    "name": "Dakota Stein",
  },
  Object {
    "controls": Array [],
    "id": 5,
    "name": "Mongo K",
  },
  Object {
    "controls": Array [],
    "id": 6,
    "name": "Dew Dong",
  },
  Object {
    "controls": Array [],
    "id": 7,
    "name": "Mort Dog",
  },
  Object {
    "controls": Array [],
    "id": 8,
    "name": "Forman Go",
  },
  Object {
    "controls": Array [],
    "id": 9,
    "name": "Dew Kelg",
  },
]
`;
